lib/tasks/route_import.rake

```diff
@@ -5,12 +5,73 @@ require 'csv'
 namespace :routes do
   task :import, [:csv_file] => :environment do |_, args|
     Route.delete_all
+    BusStopsRoute.delete_all
+    @route_hash = {}
     CSV.foreach(args[:csv_file], headers: true, col_sep: ';') do |row|
       stop = BusStop.find_by hastus_id: row['stp_identifier']
       if stop.present?
-        route = Route.find_or_create_by number: row['rte_identifier'].strip,
-                                        description: row['rte_description']
-        stop.routes << route
+        route = Route.find_or_create_by number: row['rte_identifier'].strip
+
+        direction = row['direction']
+        variant = row['variant']
+        stop_id = row['stp_identifier']
+        rank = row['stop_variant_rank']
+
+        @route_hash[route] ||= {}
+        @route_hash[route][direction] ||= {}
+        @route_hash[route][direction][variant] ||= []
+        @route_hash[route][direction][variant] << { stop_id => rank }
+      end
+    end
+
+    @stop_list = []
+    @other_variants = []
+
+    @route_hash.each do |route, directions|
```

Comment:
Procedurally I would separate "deciding what the eventual sequence should be based on various variants" and "creating the bus stop fields with that sequence" into two separate loops, probably with an intermediate data structure.

As it stands this isn't the easiest to follow.

Reply:
OHH

Reply:
So you're saying I should organize the stops per direction. And then create them.

Reply:
Did it